### DISCUSSION TITLE
Differenziazione notifiche di tipo "annotazione"

### DISCUSSION THREAD

**Discussion by antonioluberto** [italia/ansc#1079] [Supporto servizi cooperativi] [answered]
Buonasera.
Con particolare riferimento alle notifiche di tipo "annotazione" originate da specifico atto digitale, è necessario che dal set di dati restituito con le stesse sia possibile distinguere:

a) se si tratti di notifica relativa all' annotazione digitale su atto digitale di propria competenza.
Esempio: annotazione di morte che debba apporsi su atto di nascita - digitale, prodotto dal Comune di Grottaferrata - del cittadino "Rossi Mario", a seguito di atto di morte - digitale, prodotto dal Comune di Rosignano Marittimo - riferito al medesimo cittadino.

oppure

b) se si tratti di notifica relativa all'annotazione cartacea su atto che cartaceo determinata da atto digitale prodotto dal Comune Gestore.
Esempio: annotazione di morte che debba apporsi su atto di nascita - cartaceo, prodotto dal Comune di Catania - del cittadino "Rossi Mario", a seguito di atto di morte - digitale, prodotto dal Comune di Grottaferrata - riferito al medesimo cittadino.

E' necessario vi sia modo di distinguere le due tipologie di notifiche in maniera tale che:
- alla conferma di notifica che rientri nella classificazione di cui al punto a), si possa aggiornare l'archivio locale con i dati dell'annotazione; il tutto per garantire quindi che non si interrompa l'allineamento fra ANSC e la base dati comunale (di servizio), secondo principio condiviso sin dall'avvio del progetto
- si possano rendere disponibile all'utente strumenti di controllo/monitoraggio distinti e specifici , in particolare per quanto relativo alle notifiche ANPR originate nei due casi.

Rimaniamo in attesa di riscontro e naturalmente a disposizione.

**Comment by asqueglia** [comment]
Buon pomeriggio, per la distinzione delle annotazioni è possibile fare riferimento ai parametri idcanalenotifica e flagcartaceo presenti nella response delle corrispondenti notifiche, come documentato sull'ìapi del servizio delle notifiche; al momento è presente un errore sul dato flagcartaceo che verrà sanato e rilasciato ad inizio prossima settimana. Chiudiamo pertanto la discussion.

**Comment by antonioluberto** [comment]
Buongiorno.
Ancora qualche dubbio.
**Domanda 1:** è possibile che il campo indicato come "flagcartaceo" abbia in effetti nome "flageventocartaceo"?

Di seguito riportiamo quindi quanto abbiamo potuto verificare in riferimento alla valorizzazione die campi:
"idcanalenotifica"
"flageventocartaceo"
nei diversi casi possibili:

a) viene prodotto un atto "secondario" (esempio: morte) _digitale_ per un soggetto che abbia atto "primario" (nascita) _digitale_ presso lo stesso Comune Gestore (Grottaferrata): il Comune Gestore riceve la notifica di tipo "annotazione". 
Esempio:
su ambiente di pre-produzione del Comune di Grottaferrata è stato prodotto l'atto di morte 2025-5015407-200-058046 di "Rullo Angelina" 

![image](https://github.com/user-attachments/assets/0576c242-d644-40bb-9555-ce9e11478b64)

il cui atto di nascita _digitale_ è 2025-5015398-199-058046:

![image](https://github.com/user-attachments/assets/9053732b-a273-4442-8cdd-17576c222f31)

Su ambiente di pre-produzione del Comune di Grottaferrata, dal servizio R008 si ottiene quindi la notifica:
"id":"3953080"
all'interno della quale si trovano i due campi indicati (a meno del dubbio sulla descrizione di cui sopra):
![image](https://github.com/user-attachments/assets/471b15c4-e60d-4708-84f9-e00cad94df3f)

![image](https://github.com/user-attachments/assets/429c17f9-1495-4333-973a-1600d04f7b13)

b) viene prodotto un atto "secondario" (esempio: morte) _digitale_ presso il Comune A (Seregno) per un soggetto che abbia atto "primario" (nascita) _cartaceo_ presso il Comune B già in ANSC (Grottaferrata): la notifica di tipo "annotazione" va direttamente e soltanto al Comune B. 
Esempio:
su ambiente di pre-produzione del Comune di Seregno è stato prodotto l'atto di morte 2025-5015289-140-108039 di "Rolli Agnese"

![image](https://github.com/user-attachments/assets/4e83b76f-ec18-44b5-b601-c0e9bf0e6fec)

indicando, per la stesso "Rolli Agnese", atto di nascita originale cartaceo presso il Comune di Grottaferrata:

![image](https://github.com/user-attachments/assets/cc003a18-77e0-4395-aebd-830ca55c15e4)

Su ambiente di pre-produzione del Comune di Grottaferrata, dal servizio R008 si ottiene quindi la notifica:
"id":"3952548"
all'interno della quale si trovano i due campi indicati (a meno del dubbio sulla descrizione di cui sopra):

![image](https://github.com/user-attachments/assets/80dbbb27-9aa0-4f95-a98d-0b53b3592c6e)

![image](https://github.com/user-attachments/assets/4465e195-58b6-485d-b196-918236ea2a00)

c) viene prodotto un atto "secondario" (esempio: morte) _digitale_ presso il Comune A (Seregno) per un soggetto che abbia atto "primario" (nascita) _cartaceo_ presso il Comune B non ancora in ANSC (Catania), la notifica di tipo "annotazione" va ancora - e soltanto - al Comune A.
Esempio:
su ambiente di pre-produzione del Comune di Seregno è stato prodotto l'atto di morte 2025-5015300-141-108039 di "Benno Andrea":

![image](https://github.com/user-attachments/assets/8322bcca-b5ac-4751-8087-ac7949d43152)

indicando, per la stesso "Rolli Agnese", atto di nascita originale cartaceo presso il Comune di Catania (non in ANSC):

![image](https://github.com/user-attachments/assets/87c55fcd-6252-4222-8788-972259481845).

Su ambiente di pre-produzione del Comune di Seregno, dal servizio R008 si ottiene quindi la notifica:
"id":"3952576"
all'interno della quale si trovano i due campi indicati (a meno del dubbio sulla descrizione di cui sopra):

![image](https://github.com/user-attachments/assets/bcd96e7f-3b46-4740-8619-fdcee5347af9)

![image](https://github.com/user-attachments/assets/9652a928-1455-4183-974d-127dbb46f32c)

d) viene prodotto un atto "secondario" (esempio: morte) _digitale_ presso il Comune A (Seregno) per un soggetto che abbia atto "primario" (nascita) digitale presso il Comune B (Grottaferrata): la notifica di tipo "annotazione" va direttamente e soltanto al Comune B. 
Esempio:
su ambiente di pre-produzione del Comune di Seregno è stato prodotto l'atto di morte 2025-5015558-144-108039 di "Nestorov Paul"

![image](https://github.com/user-attachments/assets/e8eec02e-cb47-4c65-aa15-8c30b2bcaa2c)

il cui atto di nascita _digitale_ è 2023-1215001-254-058046 prodotto dal Comune di Grottaferrata:

![image](https://github.com/user-attachments/assets/576f805c-d294-4eae-98d6-34c78b4ec27f)

Su ambiente di pre-produzione del Comune di Grottaferrata, dal servizio R008 si ottiene quindi la notifica:
"id":"3955719"
all'interno della quale si trovano i due campi indicati (a meno del dubbio sulla descrizione di cui sopra):

![image](https://github.com/user-attachments/assets/329832b6-9ea5-41af-9f8a-fd8f738bb050)

![image](https://github.com/user-attachments/assets/658461ff-7348-4138-b411-60935f0eb2f3)

e) viene prodotto un atto "secondario" (esempio: morte) _digitale_ per un soggetto che abbia atto "primario" (nascita) _cartaceo_ presso il medesimi Comune gestore: il Comune Gestore riceve la notifica di tipo "annotazione". 
Esempio:
su ambiente di pre-produzione del Comune di Grottaferrata è stato prodotto l'atto di morte 2025-5015562-203-058046 di "Satteri Luana"

![image](https://github.com/user-attachments/assets/2513cdd5-b65b-4452-b20e-bdbef3e7200d)

indicando, per la stesso "Satteri Luana", atto di nascita originale cartaceo ancora presso il Comune di Grottaferrata:

![image](https://github.com/user-attachments/assets/826778ca-2378-444d-9f7e-ea1a1117a9bd)

Su ambiente di pre-produzione del Comune di Grottaferrata, dal servizio R008 si ottiene quindi la notifica:
"id":"3955819"
all'interno della quale si trovano i due campi indicati (a meno del dubbio sulla descrizione di cui sopra):

![image](https://github.com/user-attachments/assets/b7316b70-2203-40b3-8ccb-6a7588474dd8)

![image](https://github.com/user-attachments/assets/856ae095-152d-483f-b520-fbe468cfaa4b)


**Domanda 2**: potete gentilmente darci conferma che quello sopra rappresentato sia effettivamente il funzionamento atteso nei diversi casi esaminati e che non siano previste modifiche in merito?

Grazie, 
Antonio

**Comment by antonioluberto** [comment]
Vedi richieste al post precedente

**Comment by asqueglia** [comment, marked as answer]
Buon pomeriggio, confermiamo quanto riportato. Chiudiamo pertanto la discussion.
